### PR TITLE
fix(entity-generator): use table name instead of class name in `EntitySchema`

### LINKED ISSUE #3915: Incorrect table mapping when using `EntityGenerator` with `entitySchema`
**Describe the bug**
When using the `EntityGenerator` helper to generate the entities from existing database schema, the generated entity is mapped to wrong or non-existing table name **_especially_** when the table name has underscore (`_`). 

This may be happening for other special characters as well. 

**Stack trace**
```
...
```

**To Reproduce**
1. Create a table `Address_lines` 
  ```sql
  CREATE TABLE `Address_lines` (
  `nodeId` char(25) CHARACTER SET utf8 NOT NULL,
  `position` int(4) NOT NULL,
  `value` mediumtext COLLATE utf8mb4_unicode_ci NOT NULL,
  PRIMARY KEY (`nodeId`,`position`),
  KEY `value` (`value`(191))
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
  ```
2. setup the entity generation
  ```ts
  const orm = await init({
    // ... other config
    entitiesTs: [`./src/entities`],
    entityGenerator: {
      entitySchema: true,
      identifiedReferences: true,
    },
    type: 'mysql',
  })

  await orm.getEntityGenerator().generate({
    baseDir: path.join(__dirname, `./src/entities`),
    save: true,
  })

  await orm.close(true)
  ```
3. the following is the generated after the execusting the generation
  ```ts
    export class AddressLines {
      nodeId!: string
      position!: number
      value!: unknown
    }

  export const AddressLinesSchema = new EntitySchema({
    class: AddressLines,
    tableName: 'AddressLines', // This is wrong and should be inferred as `Address_lines`
    properties: {
      nodeId: {
        primary: true,
        fieldName: 'nodeId',
        columnType: 'char(25)',
        length: 25 
      },
      position: { primary: true, type: 'number' },
      value: {
        type: 'unknown',
        columnType: 'mediumtext',
        length: 16777215,
        index: 'value',
      },
    },
  })
  ```
**Expected behavior**
Entity class and schema files generated with corresponding table name. 
```ts
tableName: 'Address_lines',
```
**Additional context**
A possible issue could be in this line https://github.com/mikro-orm/mikro-orm/blob/master/packages/entity-generator/src/EntitySchemaSourceFile.ts#L45

**Versions**

| Dependency | Version |
| - | - |
| node | 16.x |
| typescript | 4.9.4 |
| mikro-orm | 5.6.3 |
| your-driver | mysql |

### PR DESCRIPTION
**Pull request by koneru9999**
Closes #3915

### CODE REVIEW

File: packages/entity-generator/src/EntitySchemaSourceFile.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export class EntitySchemaSourceFile extends SourceFile {
     ret += `  class: ${this.meta.className},\n`;
 
     if (this.meta.collection !== this.namingStrategy.classToTableName(this.meta.className)) {
-      ret += `  tableName: ${this.quote(this.meta.className)},\n`;
+      ret += `  tableName: ${this.quote(this.meta.tableName)},\n`;
```
Comment (B4nan) [inline packages/entity-generator/src/EntitySchemaSourceFile.ts:45]: `meta.tableName` is an alias for `meta.collection`, i'd like to use the same in the condition on line 44 as well, to reduce confusion. i dont mind using `meta.tableName` on both places

File: tests/features/entity-generator/EntityGenerator.test.ts [hunk 2]
```diff
@@ -151,6 +151,23 @@ describe('EntityGenerator', () => {
     await orm.close(true);
   });
 
+  test('table name with underscore using entitySchema [mysql]', async () => {
+    const orm = await initORMMySql('mysql', {entityGenerator: {
```
Comment (B4nan) [inline tests/features/entity-generator/EntityGenerator.test.ts:155]: is this passing the linter? :] i dont think so, the indents on following lines are also wrong

```suggestion
    const orm = await initORMMySql('mysql', { entityGenerator: {
```

**Review by B4nan — APPROVED** [review]

File: packages/entity-generator/src/EntitySchemaSourceFile.ts (hunk 1 above)
Comment (koneru9999) [reply packages/entity-generator/src/EntitySchemaSourceFile.ts:45]: thanks. applied the same check at L44 now.

File: tests/features/entity-generator/EntityGenerator.test.ts (hunk 2 above)
Comment (koneru9999) [reply tests/features/entity-generator/EntityGenerator.test.ts]: true that. i didn't run lint from my local

File: tests/features/entity-generator/EntityGenerator.test.ts [hunk 3]
```diff
@@ -151,6 +151,25 @@ describe('EntityGenerator', () => {
     await orm.close(true);
   });
 
+  test('table name with underscore using entitySchema [mysql]', async () => {
+    const orm = await initORMMySql('mysql', {
+      entityGenerator: {
+        entitySchema: true,
+        identifiedReferences: true,
+      }
```
Comment (B4nan) [inline tests/features/entity-generator/EntityGenerator.test.ts:159]:
```suggestion
      },
```

**Comment by B4nan** [comment]
Thanks!

**Pull request merged**